src/DirectGateway.php

```diff
@@ -43,6 +44,16 @@ public function setSimulatorMode($value)
         return $this->setParameter('simulatorMode', $value);
     }
 
+    public function getReferrerId()
+    {
+        return $this->getParameter('referrerId');
+    }
+
+    public function setReferrerId($value)
+    {
+        return $this->setParameter('referrerId', $value);
+    }
+
```

Comment:
I always thought this was simply the way OmniPay worked: there being one set of data for a driver, and that data can be set/get by either the driver or the message, depending on which set/get methods are available, but is always the same array or collection of data.

So there is get/set vendor here: https://github.com/thephpleague/omnipay-sagepay/blob/master/src/DirectGateway.php so that it can be set during initialisation of the gateway, but it is also here in the message abstract, otherwise the messages won't be able to access it: https://github.com/thephpleague/omnipay-sagepay/blob/master/src/Message/AbstractRequest.php

So far as I am aware, that is how it is throughout all the drivers.